### DISCUSSION TITLE
Dual Extruder with Single Nozzle and 3 active serial ports

### DISCUSSION THREAD

**Discussion by antwal** [mriscoc/Ender3V2S1#418] [General]
Hi,

I wanted to know if this firmware supported the double extruder on single nozzle from the display menu.

Rightfully I have to make several changes to the firmware from the source for the card I am using as I need 3 active serial ports and most of the firmware tested do not support both.

The board is the BTT Octopus Pro, serial -1 via usb-c, serial 1 for display, serial 2 for rpi

Thank you.

**Comment by mriscoc** [comment]
Currently there isn't support for multiple extruders.

**Comment by tbadger** [comment]
I see some code/comments about multiple extruders in the Dec. 22 code base (If I’m reading commits correctly. Is support coming, or did it get released in the new year’s build?  I’m looking to get support for using 3DCameleon, which uses a single extruder, but lets you feed up to 4 different filament for color changes, using T0, T1, T2, T3.

/**
 * M92: Set axis steps-per-unit for one or more axes, X, Y, Z, [I, [J, [K, [U, [V, [W,]]]]]] and E.
 *      (Follows the same syntax as G92)
 *
 *      With **multiple extruders use T to specify which one.**

Trent

**Reply by mriscoc** [reply]
Currently there isn't support for multiple extruders.
